fh-messaging/lib/mongo_functions/metrics_r.js

```diff
@@ -2,30 +2,70 @@
 //ts and emit are used for testing purposes
 var ts;
 var emit;
-function metrics_r(k, vals) {
-  // Take the domain and ts in the first vals entry as the reduced elements domain and ts
-  var ret = {}, i, p, tempVal;
-
-  for (i in vals) {
-    if (vals.hasOwnProperty(i)) {
-      tempVal = vals[i];
-      for (p in tempVal) {
-        if ('undefined' !== typeof ret[p]) {
-          ret[p] += tempVal[p];
-        } else {
-          ret[p] = tempVal[p];
+
+function metrics_r(k, metrics) {
+
+  function groupById(values) {
+    var groups = {};
+    for (index in values) {
+      if (values.hasOwnProperty(index)) {
+        var value = values[index];
+        if (!groups[value.id]) {
+          groups[value.id] = [];
         }
-        // this add a total field to the output, to aggregate the individual fields
-        if ('undefined' !== typeof ret['total']) {
-          ret['total'] += tempVal[p];
-        } else {
-          ret['total'] = tempVal[p];
+
+        groups[value.id].push(value);      
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)